external/fv3kube/tests/test_config.py

```diff
@@ -31,11 +31,21 @@ def test_merge_fv3config_overlays(mappings, expected):
     output = fv3kube.merge_fv3config_overlays(*mappings)
     assert output == expected
 
+def test_invalid_vertical_coorinate_file():
+
+    base_url = "/some/path"
+    timestep = "20160805.000000"
+    vertical_coordinate_file = "/some/path/"
+
+    with pytest.raises(AssertionError, match='Provided vertical coordinate file is a directory'):
+        fv3kube.c48_initial_conditions_overlay(base_url, timestep, vertical_coordinate_file)
 
 def test_c48_initial_conditions_overlay(regtest):
```

Comment:
Indeed it may make sense to add a regression test for this new functionality, though I think preserving the old regression test, demonstrating that the same call still produces the same result, would also be useful.  To do this I would use the [ability of pytest to parametrize tests](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html#parametrize).
```python
@pytest.mark.parametrize("vertical_coordinate_file", [None, "/some/path"])
def test_c48_initial_conditions_overlay(regtest, vertical_coordinate_file):
    url = "some/url"
    timestep = "20160801.000000"

    if vertical_coordinate_file is None:
        ans = fv3kube.c48_initial_conditions_overlay(url, timestep)
    else:
        ans = fv3kube.c48_initial_conditions_overlay(url, timestep, vertical_coordinate_file=vertical_coordinate_file)

    print(ans, file=regtest)
```
This preserves the regression test which calls `fv3kube.c48_initial_conditions_overlay` without a `vertical_coordinate_file` argument, which should still produce the same result as it did before.

When adding this regression test you will need to make a record of the expected output using:
```
$ pytest --regtest-reset test_config.py
```
and the add the resulting new file, which should show up in `fv3net/external/fv3kube/tests/_regtest_outputs`, to your PR.  

I think the act of parametrizing this test will also change the name of the record file for the existing regression test, so you will need to account for that as well.  The contents of that file should be identical to before, however.  To demonstrate that the contents are identical, I think you can update your PR in the following way:
```
$ pytest --regtest-reset test_config.py
$ rm _regtest_outputs/test_c48_initial_conditions_overlay[None].out
$ git mv _regtest_outputs/test_c48_initial_conditions_overlay.out _regtest_outputs/test_c48_initial_conditions_overlay[None].out
$ git add _regtest_outputs/*.out
```
By using `git mv` you are signaling to `git` that you are merely renaming a file instead of adding a completely new file and removing an old one, and so it will know in the diff to only show that the filename changed.  When updating regression tests we always want to be careful to verify that answers change only where we expect them to.  If this is done correctly you should see the regression tests passing both locally and [in our CI](https://app.circleci.com/pipelines/github/ai2cm/fv3net/14359/workflows/b1cb5125-a6b3-4ac5-8f5d-2e0b262e3b77/jobs/135241).

See [here](https://gitlab.com/uweschmitt/pytest-regtest/test) for more information on `pytest-regtest`.